docs/core/deploying/native-aot/index.md

```diff
@@ -114,24 +139,32 @@ By default, native AOT publishing produces debug information in a separate file
 
 ## Limitations of native AOT deployment
 
-Native AOT applications come with a few fundamental limitations and compatibility issues. The key limitations include:
+Native AOT apps have the following limitations:
 
-- No dynamic loading (for example, `Assembly.LoadFile`).
-- No run-time code generation (for example, `System.Reflection.Emit`).
+- No dynamic loading, for example, `Assembly.LoadFile`.
+- No run-time code generation, for example, `System.Reflection.Emit`.
 - No C++/CLI.
-- No built-in COM (only applies to Windows).
+- Windows: No built-in COM.
 - Requires trimming, which has [limitations](../trimming/incompatibilities.md).
 - Implies compilation into a single file, which has known [incompatibilities](../single-file/overview.md#api-incompatibility).
 - Apps include required runtime libraries (just like [self-contained apps](../index.md#publish-self-contained), increasing their size as compared to framework-dependent apps).
 - <xref:System.Linq.Expressions> always use their interpreted form, which is slower than run-time generated compiled code.
+- Not all the runtime libraries are fully annotated to be native AOT compatible. That is, some warnings in the runtime libraries aren't actionable by end developers.
+
+The publish process analyzes the entire project and its dependencies for possible limitations. Warnings are issued for each limitation the published app may encounter at run time.
 
-The publish process analyzes the entire project and its dependencies and produces warnings whenever the limitations of the published application might be encountered at run time.
+### OS specific limitations
+
+### [.NET 7](#tab/net7)
+
+- Should be targeted for console type apps. ASP.NET Core is ***not*** supported.
+- Limited diagnostic support for debugging and profiling.
+
+### [.NET 8+](#tab/net8plus)
 
-The first release of native AOT in .NET 7 has some more limitations:
+Support for some ASP.NET Core features. For more information, see [ASP.NET Core support for native AOT](/aspnet/core/fundamentals/native-aot/?view=aspnetcore-8.0&preserve-view=true).
```

Comment:
The diagnostics limitations are still in place for .NET 8 - some of them were fixed, but I would not call it done at all.

Reply:
#36529